### ISSUE TITLE
Cannot set OomKillDisable through docker api /containers/{id}/update

### ISSUE THREAD

**Issue by dorazwt** [moby/moby#42588]
**Description**
I start a container without --oom-kill-disable, then I POST {"OomKillDsiable": true} to the docker api "/containers/{id}/update". I got a 200 response code, but it does not work. According to the [api doc](https://docs.docker.com/engine/api/v1.40/#operation/ContainerUpdate), it should work.
Then I tried starting a container with --oom-kill-disable + POST {"OomKillDsiable": false}, it still has no effects.

**Steps to reproduce the issue:**
1. start a container ```docker run -it xxx:x```
2. POST {"OomKillDsiable": false} to the docker api "/containers/{id}/update", get a 200 response 
3. check the prop via docker api "/containers/{id}/json"

**Describe the results you received:**
POST /containers/xxx/update response: {"Warnings": ["OOM killer is disabled for the container xxxxxx"]}
GET /containers/xxx/json response: {"OomKillDisable": false}

**Describe the results you expected:**
GET /containers/xxx/json response: {"OomKillDisable": true}

**Output of `docker version`:**

```
Client: Docker Engine = Community
 Version: 19.03.12
 API version: 1.40
 Go version: go1.13.10
 Git commit: 48a66213fe
 Built: Mon Jun 22 15:45:44 2020
 Os/Arch: linux/amd64
 Experimental: false

Server: Docker Engine - Community
 Engine:
  Version: 19.03.12
  API version: 1.40 (minimum version 1.12) 
  Go version: go1.13.10
  Git commit: 48a66213fe
  Built: Mon Jun 22 15:44:15 2020
  Os/Arch: linux/amd64
  Experimental: false
 containerd:
  Version: 1.2.13
 docker-init:
  Version: 0.18.0
```

**Comment by Rid** [comment]
Can you post `docker info`? `OomKillDsiable` does not work on cgroups v2.

**Comment by sam-thibault** [comment]
I don't see any recent activity on this issue.  I will close it as stale.

**Comment by thaJeztah** [comment]
There may be some room for improvement here. On a machine running with cgroups v2;

```bash
docker info | grep Cgroup
 Cgroup Driver: systemd
 Cgroup Version: 2
```

`docker run --help` does not indicate it's _not_ supported (I know we hide flags for other unsupported options);

```bash
docker run --help | grep oom
      --oom-kill-disable               Disable OOM Killer
      --oom-score-adj int              Tune host's OOM preferences (-1000 to 1000)
```

But we do warn when trying to use it for a container;

```bash
docker run --memory=10m --oom-kill-disable --rm alpine
WARNING: Your kernel does not support OomKillDisable. OomKillDisable discarded.
```


the `docker info` output also doesn't contain a warning that it's not supported (we do this for other options to make it visible), and I just recall that this was disabled at some point to not print the warning for Cgroup v2;
https://github.com/moby/moby/blob/487fa4a469b4fc77e1706f14acbd97c06383e754/daemon/info_unix.go#L96-L105

See:

- https://github.com/moby/moby/pull/41894
- https://github.com/moby/moby/issues/41790

I'm slightly on the fence if we should still print _a_ warning (but perhaps rephrase it to indicate that it's not supported on cgroups v2); OTOH, cgroups v2 is the default now for most distros, so it might be "noisy".

- Update the `--oom-kill-disable` option to include that it's "cgroup v1 only" (easy fix)
- Hide the flag if we detected the daemon cannot support it (requires some changes in the info returned by `/_ping` - not too complicated, but we need to think this over)
- Change how we present the supported options; the `/info` endpoint already contains the information, but we (I think) currently only use it to propagate the `Warnings[]` array; we could change the output to show it as supported features (supported / not supported) instead. This is from the API response; 

    ```json
      "MemoryLimit": true,
      "SwapLimit": true,
      "CpuCfsPeriod": true,
      "CpuCfsQuota": true,
      "CPUShares": true,
      "CPUSet": true,
      "PidsLimit": true,
      "IPv4Forwarding": true,
      "BridgeNfIptables": true,
      "BridgeNfIp6tables": true,
      "Debug": false,
      "NFd": 45,
      "OomKillDisable": false,
    ```
